File: php/src/Model/StatsTimeseriesResult.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

use Fivexer\SDK\Internal\Json;

/**
 * Historical throughput over a window.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class StatsTimeseriesResult
{
    public function __construct(
        public readonly string $from,
        public readonly string $to,
        public readonly string $bucket,
        /** @var list<StatsTimeseriesBucket> */
        public readonly array $buckets,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            from: (string) ($data['from'] ?? ''),
            to: (string) ($data['to'] ?? ''),
            bucket: (string) ($data['bucket'] ?? ''),
            buckets: Json::parseEach($data, 'buckets', [StatsTimeseriesBucket::class, 'fromArray']),
        );
    }
}

File: java/src/main/java/io/fivexer/sdk/model/StatsTimeseriesResult.java
package io.fivexer.sdk.model;

import java.util.List;

/**
 * Historical throughput over a window.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class StatsTimeseriesResult {
    private String from;
    private String to;
    private String bucket;
    private List<StatsTimeseriesBucket> buckets;

    public String getFrom() { return from; }
    public String getTo() { return to; }
    public String getBucket() { return bucket; }
    public List<StatsTimeseriesBucket> getBuckets() { return buckets; }
}
